numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -1008,11 +1022,17 @@ PyArray_MatrixProduct2(PyObject *op1, PyObject *op2, PyArrayObject* out)
         return NULL;
     }
 
+    npy_clear_floatstatus();
+
 #if defined(HAVE_CBLAS)
     if (PyArray_NDIM(ap1) <= 2 && PyArray_NDIM(ap2) <= 2 &&
             (NPY_DOUBLE == typenum || NPY_CDOUBLE == typenum ||
              NPY_FLOAT == typenum || NPY_CFLOAT == typenum)) {
-        return cblas_matrixproduct(typenum, ap1, ap2, out);
+        PyObject *res = cblas_matrixproduct(typenum, ap1, ap2, out);
+        if (check_fperr("dot") < 0) {
+            return NULL;
```

Comment:
Just to note, this is also incorrect of course:  You would have to throw away `res` here, but you are leaking it.